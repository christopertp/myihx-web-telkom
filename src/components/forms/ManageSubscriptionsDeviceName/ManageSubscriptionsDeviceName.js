import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import { useSelector } from 'react-redux';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import TextField from '../../fields/Text';
import styles from './styles.scoped.css';

export default function ManageSubscriptionsDeviceName(props) {
  const { handleSubmit, invalid, handleCancel, message } = props;
  const { isLoading } = useSelector((s) => s.manageSubscriptions);
  const inputProps = {
    autoFocus: true,
    label: 'DEVICE NAME',
    placeholder: 'Masukkan Nama Device',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <h3>Change device name</h3>
      <Field component={TextField} inputProps={inputProps} name="deviceName" />
      {message && <small className={styles.error}>{message}</small>}
      <footer>
        <Button onClick={handleCancel} type="submit" variant="bordered">
          Cancel
        </Button>
        <Button disabled={invalid} type="submit">
          {isLoading.s ? <Spinner /> : 'Save'}
        </Button>
      </footer>
    </form>
  );
}

ManageSubscriptionsDeviceName.defaultProps = {
  handleCancel: () => {},
  handleSubmit: () => {},
  invalid: false,
  message: '',
};

ManageSubscriptionsDeviceName.propTypes = {
  handleCancel: PropTypes.func,
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
  message: PropTypes.string,
};
