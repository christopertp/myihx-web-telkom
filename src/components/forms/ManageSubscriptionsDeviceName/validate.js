export default function validate(values) {
  const { deviceName } = values;

  return {
    deviceName: !deviceName ? 'Harus diisi' : '',
  };
}
