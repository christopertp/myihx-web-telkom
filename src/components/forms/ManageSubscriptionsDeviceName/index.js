import { reduxForm } from 'redux-form';
import ManageSubscriptionsDeviceName from './ManageSubscriptionsDeviceName';
import validate from './validate';

export default reduxForm({
  initialValues: {
    deviceName: '',
  },
  form: 'manageSubscriptionsDeviceName',
  validate,
})(ManageSubscriptionsDeviceName);
