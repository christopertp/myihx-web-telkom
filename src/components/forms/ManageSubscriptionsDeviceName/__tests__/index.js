import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import ManageSubscriptionsDeviceName from '../ManageSubscriptionsDeviceName';

useSelector.mockImplementation(fn => {
  fn(({ manageSubscriptions: '' }));
  return { isLoading:{ s: false }, message: '' };
});

describe('src/components/forms/ManageSubscriptionsDeviceName', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ManageSubscriptionsDeviceName />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading:{ s: true }, }));
    ManageSubscriptionsDeviceName.defaultProps.handleSubmit();
    ManageSubscriptionsDeviceName.defaultProps.handleCancel();
    const result = ManageSubscriptionsDeviceName({
      ...ManageSubscriptionsDeviceName.defaultProps,
      message:'test error'
    });
    const footer = result.props.children[3];
    expect(footer.props.children[1].props.children.type).toBe(Spinner);
  });
});
