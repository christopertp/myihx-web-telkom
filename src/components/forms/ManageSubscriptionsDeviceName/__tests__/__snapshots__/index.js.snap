// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ManageSubscriptionsDeviceName render 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Change device name
  </h3>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "autoFocus": true,
        "label": "DEVICE NAME",
        "placeholder": "Masukkan Nama Device",
      }
    }
    name="deviceName"
  />
  
  <footer>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="submit"
      variant="bordered"
    >
      Cancel
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to=""
      type="submit"
      variant="red"
    >
      Save
    </Button>
  </footer>
</form>
`;
